Support for OpenTracing 0.33.0

agarbutt opened the pull request:
Support for #18

jeffalder commented:
Hi @agarbutt , Thanks for the contribution! I'm going to take a look at this.

jeffalder commented on src/main/java/com/newrelic/opentracing/aws/TracingRequestHandler.java:
```diff
@@ -42,16 +43,18 @@ default Output handleRequest(Input input, Context context) {
         throw throwable;
       }
+    } finally {
+      span.finish();
```
am I the only one annoyed that `Span` doesn't implement `Closeable`, so you could _really_ use try-with-resources?

agarbutt replied:
yeah, the choice made to move away from making the Span Auto-Closable was a surprise to me.

jasonjkeller commented:
We should also update the AWS Lambda Tracer to support OpenTracing 0.33 in conjunction with the this PR for the SDK. https://newrelic.atlassian.net/browse/JAVA-5898

agarbutt commented on src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java:
```diff
@@ -0,0 +1,70 @@
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+
+public class EnhancedSpanBuilder {
```
Thanks. I had / have some other ideas, but if you're moving to OpenTelemetry, i'd rather get this minimally supported and then invest there. 👍

jeffalder replied:
Agreed

agarbutt commented on src/main/java/com/newrelic/opentracing/aws/SpanUtil.java:
```diff
@@ -6,29 +6,29 @@
+        .optionallyWithTag(
+            "aws.lambda.eventSource.arn", EventSourceParser.parseEventSourceArn(input))
+        .optionallyWithTag(
+            "aws.lambda.coldStart", TracingRequestHandler.isColdStart.getAndSet(false))
```
Great catch! This is a bad refactoring. See my updated commit.

agarbutt commented on src/main/java/com/newrelic/opentracing/aws/EnhancedSpanBuilder.java:
```diff
@@ -0,0 +1,70 @@
+package com.newrelic.opentracing.aws;
```
Done.

agarbutt commented:
Thanks for the feedback @jeffalder. I believe I have addressed the concerns you raised. 

I'd be happy to help contribute, if that would be helpful, @jasonjkeller

jeffalder commented:
We'll have to ship an update to the other side of this to match up OT versions.
